pyteal/ast/abi/reference_type.py

```diff
@@ -27,8 +29,11 @@ class Account(Uint):
     def __init__(self) -> None:
         super().__init__(AccountTypeSpec())
 
-    def deref(self) -> Expr:
-        return Txn.accounts[self.stored_value.load()]
+    def get(self) -> Expr:
```

Comment:
I like this improvement, but it brings to light some contradictions:
* `Account` inherits from `Unit`, but returning a non-integer here breaks the behavior you'd expect from a `Uint`.
* For `Application` and `Asset`, if you were to inspect their type specs, they claim to return 8 bit integers, but in reality they're really 64 bits.

Would it possible for reference types to not inherit from Uint for these reasons?

Reply:
you mean create a new ReferenceTypeSpec and ReferenceType and have them inherit from that?

Reply:
Yeah that would work

Reply:
I'm not quite sure how to handle this tbh, the over the wire encoding is a uint8 but get returns either uint64 or a bytestring. 

I'm not sure what to do for things like byte_length_static for example, should it be the over the wire encoding or the return type for `get`?

Reply:
https://github.com/algorand/pyteal/pull/383 moving this set of changes to new pr